### PR TITLE
ObjectReflectionCache - Added support for IReadOnlyDictionary as Expando-object

### PR DESCRIPTION
**Pull request by snakefoot**
Also optimized performance by reducing allocation for empty dictionary.

### CODE REVIEW

File: src/NLog/Internal/ObjectReflectionCache.cs [hunk 1]
```diff
@@ -137,13 +137,24 @@ private static bool TryExtractExpandoObject(Type objectType, out ObjectPropertyI
...
+                            return true;
+                        }
+#if NET4_5
+                        if (interfaceType.GetGenericTypeDefinition() == typeof(IReadOnlyDictionary<,>))
+                        {
+                            var dictionaryEnumerator = (IDictionaryEnumerator)Activator.CreateInstance(typeof(DictionaryEnumerator<,>).MakeGenericType(interfaceType.GetGenericArguments()));
+                            propertyInfos = new ObjectPropertyInfos(null, new[] { new FastPropertyLookup(string.Empty, TypeCode.Object, (o, p) => dictionaryEnumerator.GetEnumerator(o)) });
+                            return true;
+                        }
+#endif
```
Comment (304NotModified) [inline src/NLog/Internal/ObjectReflectionCache.cs:157]: Would it be possible to use reduce the code duplication?

Comment (snakefoot) [reply src/NLog/Internal/ObjectReflectionCache.cs:145]: Reduced code duplication

**Comment by snakefoot** [comment]
Sorry. I was just playing with code-coverage.

**Comment by 304NotModified** [comment]
cool!

One thing, the PR name. I don't think end users could understand this. I don't know how to rename this one.

**Comment by 304NotModified** [comment]
> One thing, the PR name. I don't think end users could understand this. I don't know how to rename this one.

Any good ideas?

**Comment by snakefoot** [comment]
> Any good ideas?

Nope. It does as the title say. Allow expando objects to work without implementing `IDictionary` or `IDictionary<TKey,TValue>`

**Pull request merged**